File: JavaScript/6js 2/solve_consonant.js
/*
|--------------------------------------------------------------------------
| Consonant value
|--------------------------------------------------------------------------
|
*/

function solve(s) {
   const isConsonant = (char) => !"aeiou".includes(char);
   let res = 0;
   let temp = 0;
   for (const char of s) {
      if (isConsonant(char)) {
         temp += char.charCodeAt(0) - 96;
         res = Math.max(res, temp);
      } else {
         temp = 0;
      }
   }
   return res;
}


function solve(s) {
   return Math.max(
      ...s.split(/[aeiou]+/).map((e) => [...e].reduce((a, b) => a + b.charCodeAt() - 96, 0))
   );
}


function solve(s) {
   return Math.max(
      ...s.match(/[^aeiou]+/g).map((x) => [...x].reduce((s, v) => s + v.charCodeAt() - 96, 0))
   );
}


console.log(solve("zodiac"), 26);
console.log(solve("chruschtschov"), 80);
console.log(solve("khrushchev"), 38);
console.log(solve("strength"), 57);
console.log(solve("catchphrase"), 73);
console.log(solve("twelfthstreet"), 103);
console.log(solve("mischtschenkoana"), 80);
console.log(solve("az"), 26);

File: Python/6py 2/solve_consonant.py
"""
|--------------------------------------------------------------------------
| Consonant value
|--------------------------------------------------------------------------
|
"""

def solve(s):
   res, temp = 0, 0
   for char in s:
      if char not in 'aeiou':
         temp += ord(char) - 96
         res = max(res, temp)
      else: temp = 0
   return res


def solve(s):
   return max(sum(ord(c) - ord('a') + 1 for c in p) for p in s.translate(str.maketrans('aeiou', '     ')).split())


from re import split
def solve(s):
   return max(sum(ord(c)-96 for c in subs) for subs in split('[aeiou]+', s))


print(solve("cozy"), 51)
print(solve("xyzzy"), 126)
print(solve("zodiac"), 26)
print(solve("chruschtschov"), 80)
print(solve("khrushchev"), 38)
print(solve("strength"), 57)
print(solve("catchphrase"), 73)
print(solve("twelfthstreet"), 103)
print(solve("mischtschenkoana"), 80)